src/SIL.LCModel.Core/Text/TextExtensions.cs

```diff
@@ -355,6 +355,7 @@ public static bool IsCharWordForming(this ITsString tss, int ich, ILgWritingSyst
 			if (ich < 0 || ich >= tss.Length)
 				throw new ArgumentOutOfRangeException("ich");
 
+			if (char.IsLowSurrogate(tss.Text[ich])) --ich;
```

Comment:
please move the --ich to the next line.

If there is already a unit test that covers this function we should add a new test to document this fix.

Reply:
We moved --ich to next line.